scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -225,10 +227,20 @@ def image(self) -> str:
         """Environment image."""
         return self._yaml.get('image')
 
+    @property
+    def type(self) -> str:
+        """Type of a particular asset.
+
+        For eg:
+            `custom_model` or `mlflow_model` for a model asset.
+            `command`, `pipeline` etc. for a component asset.
+        """
+        return self._yaml.get("type", "command")

```

Comment:
This has a default value that wouldn't be correct for other spec files. Consider not providing a default.
```suggestion
        return self._yaml.get('type')
```

Reply:
v2 component spec has command by default set if no type provided. Such are valid spec files. Are we going to enforce type for spec's yaml?

Reply:
At some point I think we'll need to create subclasses of Spec to handle Components, Environments, etc. but for now it's just a way to deal with the YAML file in a somewhat object-oriented way. Let's go with no default value here and allow code that calls this function to handle `None` values as `command` because it has more context.

Reply:
got it thanks